graph/telemetry/istio/appender/mesh_check.go

```diff
@@ -95,15 +97,9 @@ func (a *MeshCheckAppender) applyMeshChecks(trafficMap graph.TrafficMap, globalI
 	}
 }
 
-// namespaceOk returns true if the namespace in question is the current appender namespace or any of the graph namespaces
-func (a *MeshCheckAppender) namespaceOK(namespace string, namespaceInfo *graph.AppenderNamespaceInfo) bool {
-	if namespace == namespaceInfo.Namespace {
-		return true
-	}
-	for _, ns := range a.AccessibleNamespaces {
-		if namespace == ns.Name {
-			return true
-		}
-	}
-	return false
+// nodeOK returns true if we have access to its workload info
+func (a *MeshCheckAppender) nodeOK(node *graph.Node, namespaceInfo *graph.AppenderNamespaceInfo) bool {
+	key := fmt.Sprintf("%s:%s", node.Cluster, node.Namespace)
```

Comment:
alarm bells go off when I see the same Sprintf everywhere building some formatted string (cluster, followed by colon, followed by namespace).

Consider creating a common utility function that everyone uses so you don't have these Sprintfs all over the place (especially if in the future that syntax needs to change for some reason).